terraform/layers/app/s3.tf

```diff
@@ -0,0 +1,14 @@
+locals {
+  resource_name_prefix = "${var.project_name}-${var.environment}"
+}
+
+data "aws_s3_bucket" "s3_webfiles_bucket" {
+  bucket = "${local.resource_name_prefix}-webfiles"
+}
+
+resource "aws_s3_object" "index_html_file" {
+  bucket = "${local.resource_name_prefix}-webfiles"
```

Comment:
As we've used the above data block to retrieve the S3 bucket, we should reference the bucket directly as it creates a dependency when running terraform plan i.e this resource is executed after the data block has checked the bucket exists. So something like data.aws_s3_bucket.s3_webfiles_bucket.id (double check the docs for exact syntax)